test/framework/src/main/java/org/elasticsearch/test/disruption/LongGCDisruption.java

```diff
@@ -228,23 +237,31 @@ protected boolean stopNodeThreads(Set<Thread> nodeThreads) {
                 if (thread.isAlive() && nodeThreads.add(thread)) {
                     liveThreadsFound = true;
                     logger.trace("stopping thread [{}]", threadName);
-                    thread.suspend();
-                    // double check the thread is not in a shared resource like logging. If so, let it go and come back..
                     boolean safe = true;
```

Comment:
I wonder if it'd be save is we defaulted to safe = false and only set it to true at the bottom of the try block. That way if we throw anything in the try block we'd resume the thread.